includes/Checker/Runtime_Environment_Setup.php

```diff
@@ -48,6 +48,10 @@ static function () use ( $permalink_structure ) {
 				}
 			);
 
+			if ( ! isset( $_SERVER['HTTP_HOST'] ) ) {
+				$_SERVER['HTTP_HOST'] = get_option( 'siteurl' );
```

Comment:
It seems like Behat issue, found one https://groups.google.com/g/behat/c/n4ZYbpIRNVo/m/Q34pjvbA1ikJ that we are fixing here as `wp_guess_url()` called in `populate_options()` at first place in WP and Behat not set proper server variables.

See Behat Server variable:
```php
Array(
    [SERVER_SOFTWARE] => 
    [REQUEST_URI] => 
    [HOME] => /tmp/wp-cli-home
    [BEHAT_SCENARIO_TITLE] => Perform runtime check for multi-file plugin
    [BEHAT_FEATURE_TITLE] => Test that the WP-CLI command works.
    [BEHAT_PROJECT_DIR] => /home/runner/work/plugin-check/plugin-check
    [PATH] => bin:vendor/bin:/home/runner/.composer/vendor/bin:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
    [PWD] => /tmp/wp-cli-test-run--66bdf869cd91a9.96760688
    [BEHAT_RUN] => 1
    [PHP_SELF] => bin/wp
    [SCRIPT_NAME] => bin/wp
    [SCRIPT_FILENAME] => bin/wp
    [PATH_TRANSLATED] => bin/wp
    [DOCUMENT_ROOT] => /tmp/wp-cli-test-run--66bdf869cd91a9.96760688
    [REQUEST_TIME_FLOAT] => 172[37](https://github.com/mukeshpanchal27/plugin-check/actions/runs/10403977362/job/28811496240?pr=12#step:13:38)25930.8629
    [REQUEST_TIME] => 1723725930
    [argv] => Array
        (
            [0] => bin/wp
            [1] => plugin
            [2] => check
            [3] => foo-sample
            [4] => --fields=code,type
            [5] => --format=csv
            [6] => --require=./wp-content/plugins/plugin-check/cli.php
        )

    [argc] => 7
    [SERVER_PROTOCOL] => HTTP/1.0
    [HTTP_USER_AGENT] => 
    [REQUEST_METHOD] => GET
    [REMOTE_ADDR] => 127.0.0.1
)
```

The propose changed set full URL? ex. `http://localhost:8888` that is wrong. Instead could we update.

```suggestion
				$site_url             = get_option( 'siteurl' );
				$_SERVER['HTTP_HOST'] = preg_replace( '#^https?://#', '', rtrim( $site_url, '/' ) );
```